Userland/Applications/NetworkSettings/NetworkSettingsWidget.cpp

```diff
@@ -134,27 +138,64 @@ void NetworkSettingsWidget::on_switch_enabled_or_dhcp()
 
 void NetworkSettingsWidget::apply_settings()
 {
-    auto config_file = Core::ConfigFile::open_for_system("Network", Core::ConfigFile::AllowWriting::Yes).release_value_but_fixme_should_propagate_errors();
+    auto result = apply_settings_impl();
+    if (result.is_error()) {
+        GUI::MessageBox::show_error(window(), result.release_error().string_literal());
+        return;
+    }
+}
+
+ErrorOr<void> NetworkSettingsWidget::apply_settings_impl()
+{
+    auto json = TRY(create_settings_object());
+    if (json.is_empty())
+        return {};
+
+    auto wpipefds = TRY(Core::System::pipe2(O_CLOEXEC));
+
+    posix_spawn_file_actions_t file_actions;
+    posix_spawn_file_actions_init(&file_actions);
+    posix_spawn_file_actions_adddup2(&file_actions, wpipefds[0], STDIN_FILENO);
+
+    char const* argv[] = { "network-settings-helper", nullptr };
+    (void)TRY(Core::System::posix_spawnp("network-settings-helper"sv, &file_actions, nullptr, const_cast<char**>(argv), environ));
+
+    posix_spawn_file_actions_destroy(&file_actions);
+
+    TRY(Core::System::close(wpipefds[0]));
```

Comment:
Are we leaking `wpipefds[1]` here? We'll also leak `wpipefds[0]` if the above `posix_spawnp` fails, as well as the memory allocated by `posix_spawn_file_actions_init`. We can use `ScopeGuard` to protect these, something like (untested):
```c++
    auto wpipefds = TRY(Core::System::pipe2(O_CLOEXEC));
    ScopeGuard guard_fd1 { [&] { close(wpipefds[1]); } };
    {
        posix_spawn_file_actions_t file_actions;
        posix_spawn_file_actions_init(&file_actions);
        posix_spawn_file_actions_adddup2(&file_actions, wpipefds[0], STDIN_FILENO);

        ScopeGuard guard_fd0_and_file_actions { [&] { 
            posix_spawn_file_actions_destroy(&file_actions);
            close(wpipefds[0]); 
        } };

        char const* argv[] = { "network-settings-helper", nullptr };
        (void)TRY(Core::System::posix_spawnp("network-settings-helper"sv, &file_actions, nullptr, const_cast<char**>(argv), environ));
    }

    auto outfile = TRY(Core::File::adopt_fd(wpipefds[1], Core::File::OpenMode::Write));
    TRY(outfile->write_until_depleted(json.serialized<StringBuilder>().bytes()));

```

Reply:
Can I get this to work with `Core::System::close` instead of `close`?

The lamda is not happy about the `TRY` usage..

Reply:
I opted to use `close` for the moment.